iphone/Maps/Classes/MapDownloader/MWMBaseMapDownloaderViewController.mm

```diff
@@ -501,6 +501,11 @@ - (IBAction)longPress:(UILongPressGestureRecognizer *)sender
 
 - (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex
 {
+  if (buttonIndex >= actionSheet.numberOfButtons)
```

Comment:
Если это был не критичный креш, тем более не нужно было его исправлять. А если уж исправлять, то обязательно с подробным комментарием.

Вероятность бага в системной апишке на порядки ниже вероятности бага в нашем коде. Этот креш мог быть следствием другого бага, который вот это “замазывание” потенциально спрятало.

> On Apr 8, 2016, at 16:09, Vlad Mihaylenko <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/MapDownloader/MWMBaseMapDownloaderViewController.mm <https://github.com/mapsme/omim/pull/2779#discussion_r59021684>:
> 
> > @@ -501,6 +501,11 @@ - (IBAction)longPress:(UILongPressGestureRecognizer *)sender
> >  
> >  - (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex
> >  {
> > +  if (buttonIndex >= actionSheet.numberOfButtons)
> этот код показался мне достаточно хорошим, чтобы его помержить.
> 
> креш был только на 7ых осях, что уже делает его не критичным. У нас на 7ой оси многое не работает или работает неверно. Причем, чаще всего, что апи эпла работает неверно (или не работает). Но оставлять лишние креши мне не кажется разумным.
> можно спросить у Артема, оставлять нам креши или фиксить. Просто времени разбираться, почему эпловая апишка выдает поведение, которое не соответствует ее документации, на тот момент вообще не было.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2779/files/362adb4fcf0abf0f56f27377d9292e1ce953f00f#r59021684>